src/devices/visonic_powercode.c

```diff
@@ -0,0 +1,124 @@
+/** @file
+    Decoder for Visonic Powercode devices. Tested with an MCT-302.
+
+    Copyright (C) 2020 Maxwell Lock
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+ */
+
+/**
+The device uses OOK PWM encoding, short pulse 400us long pulse 800us, and repeats 6 times
+You can use a flex decoder -X 'n=visonic_powercode,m=OOK_PWM,s=400,l=800,r=5000,g=900,t=160,y=0'
+
+Powercode packet structure is 37 bits. 4 examples follow
+
+              s addr                       data     cksm
+              1 01101111 01000111 01110000 10001100 1001 - magnet near, case open
+              1 01101111 01000111 01110000 11001100 1101 - magnet away, case open
+              1 01101111 01000111 01110000 00001100 0001 - magnet near, case closed
+              1 01101111 01000111 01110000 01001100 0101 - magnet away, case closed
+              | |                        | |||||||| |  |
+     StartBit_/ /                        / |||||||| \__\_checksum, XOR of preceding nibbles
+     DeviceID__/________________________/  ||||||||
+                                           ||||||||
+                                    Tamper_/||||||\_Repeater
+                                      Alarm_/||||\_Spidernet
+                                     Battery_/||\_Supervise
+                                         Else_/\_Restore
+
+1 bit start bit
+3 byte(24 bit) device ID
+1 byte data
+1 nibble (4 bit) checksum
+
+Checksum is a londitudinal redundancy check of the 4 bytes containing the device ID and data. Bytes are split into nibbles. 1st bit of each nibble is XORed and result is 1st bit of checksum, then the same for the 2nd, 3rd and 4th bits. To be implemented.
```

Comment:
longitudinal

also, consider splitting into multiple lines for readability (e.g. github mobile doesnt line wrap automatically)

Reply:
fixed